src/freenet/io/comm/UdpSocketHandler.java

```diff
@@ -64,6 +64,7 @@ public UdpSocketHandler(int listenPort, InetAddress bindto, Node node, long star
 		try {
 			// Exit reasonably quickly
 			_sock.setReuseAddress(true);
+			_sock.setTrafficClass(node.getTrafficClass().value);
```

Comment:
From the `setTrafficClass` Javadoc:
> Setting bits in the precedence field [the 3 most significant bits, red.] may result in a `SocketException` indicating that the operation is not permitted. 

Failing to set the traffic class is not a fatal error, but failure can be expected on some systems because of the above. This instruction should have its own try/catch that at most logs a warning if setting the traffic class fails, but should certainly not throw a fatal `RuntimeException` as per the current try/catch.